src/render/svgUtils.ts

```diff
@@ -0,0 +1,162 @@
+    }
+
+    if (iconData) {
+      const iconSVG = this.canvas.path(iconData)
+      iconSVG.fill(color)
+      const iconBBox = iconSVG.bbox()
+      const targetHeight = this.buttonRes.height / 1.3
+      const targetWidth = this.buttonRes.width / 1.3
+      const scaleFactor = Math.min(targetHeight / iconBBox.height, targetWidth / iconBBox.width)
+      iconSVG.center(this.buttonRes.width / 2, this.buttonRes.height / 2)
+      iconSVG.scale(scaleFactor)
+    }
+
+    const outerSVG = this.canvas.svg().replaceAll('%', '&#37;')
+    this.canvas.clear()
+    return outerSVG
+  }
+
+  private generateButtonSVG(
+    labels: string[],
+    mdiIconName: Nullable<string>,
+    iconColor: string,
+    isAction = false,
+    isMultiAction = false
+  ) {
+    let iconData: Nullable<string> = null
+    if (mdiIconName) {
+      // @ts-expect-error dynamic import of icon based on name
+      // eslint-disable-next-line import/namespace
+      iconData = Mdi[this.toPascalCase(mdiIconName)] as Nullable<string>
+    }
+
+    if (iconData) {
+      const iconSVG = this.canvas.path(iconData)
+      iconSVG.fill(iconColor)
+      const iconBBox = iconSVG.bbox()
+      const targetHeight = this.buttonRes.height / 2 / 1.2
+      const targetWidth = this.buttonRes.width / 1.3
+      const scaleFactor = Math.min(targetHeight / iconBBox.height, targetWidth / iconBBox.width)
+      iconSVG.center(this.buttonRes.width / 2, this.buttonRes.height / 4)
+      iconSVG.scale(scaleFactor)
+    }
+
+    if (isAction) {
+      const color = isMultiAction ? '#3e89ff' : '#62ff65'
+      this.canvas
+        .circle(30)
+        .center(this.buttonRes.width - 1, 0)
+        .fill(color)
+    }
+
+    let currentLineNumber = 0
+    for (let i = 0; i < labels.length; i++) {
+      const lines = labels[i].split('\n')
+      for (let i = currentLineNumber; i < lines.length + currentLineNumber; i++) {
+        this.drawText(lines[i - currentLineNumber], i)
+      }
+      currentLineNumber += lines.length
+    }
+
+    // Debug Grid
+    // this.canvas.line(this.halfRes.width, 0, this.halfRes.width, this.buttonRes.height).stroke("#FFFFFF")
+    // this.canvas.line(this.halfRes.width / 2, 0, this.halfRes.width / 2, this.buttonRes.height).stroke("#FFFFFF")
+    // this.canvas.line(this.halfRes.width * 1.5, 0, this.halfRes.width * 1.5, this.buttonRes.height).stroke("#FFFFFF")
+    // this.canvas.line(0, this.halfRes.height, this.buttonRes.width, this.halfRes.height).stroke("#FFFFFF")
+    // this.canvas.line(0, this.halfRes.width / 2, this.buttonRes.width, this.halfRes.height / 2).stroke("#FFFFFF")
+    // this.canvas.line(0, this.halfRes.width * 1.5, this.buttonRes.width, this.halfRes.height * 1.5).stroke("#FFFFFF")
+
+    const outerSVG = this.canvas.svg().replaceAll('%', '&#37;')
+    this.canvas.clear()
+    return outerSVG
+  }
+
+  private drawText(text: string, lineNr: number) {
+    const escapedText = encodeURIComponent(text)
```

Comment:
@cgiesche It looks like this line is the one causing a problem... but it's not new 😕.

Removing it makes it render correctly.  Maybe just a difference between the old and new SVG libraries.